py3status/modules/mpd_status.py

```diff
@@ -110,12 +110,30 @@ class Py3status:
     state_play = '[play]'
     state_stop = '[stop]'
 
+    def _connection(self):
+        try:
+            if self.connection is None:
+                self.connection = MPDClient()
+                self.connection.connect(host=self.host, port=self.port)
+                if self.password:
+                    self.connection.password(self.password)
+            return self.connection
+        except (socket.error, ConnectionError, CommandError) as e:
+            self.connection = None
+            raise e
+
+    def _mpd_disconnect(self):
+        self._connection().disconnect()
+        self.connection = None
+
     def post_config_hook(self):
         # Convert from %placeholder% to {placeholder}
         # This is not perfect but should be good enough
         if not self.py3.get_placeholders_list(self.format) and '%' in self.format:
             self.format = re.sub('%([a-z]+)%', r'{\1}', self.format)
             self.py3.log('Old % style format DEPRECATED use { style format')
+        # class variables:
+        self.connection = None
```

Comment:
Rename `self.connection` to `self.client` to remove the confusion between `self.connection` and `self._connection`.

Reply:
that one i do find strange. _we_ are the client, what gets stored there is the server. but since i'll change the name of _connection(), that shouldn't be an issue anyways!

ps: i do appreciate the feedback - keep it comin'!

Reply:
`client library` http://pythonhosted.org/python-mpd2/topics/getting-started.html

Reply:
then the example is wrong!!!11 :P (changes made)

Reply:
I'll test this... You go and make us a new module.   .__.